src/ConstructiveSolidGeometry/VolumePrimitives/RegularPrism.jl

```diff
@@ -0,0 +1,90 @@
+print(io::IO, rp::HexagonalPrism{T, TR, TZ})  where {T,TR,TZ} = print(io, "HexagonalPrism{$(T), $(TR), $(TZ)}($(rp.r), $(rp.z))")
+
+
+# Constructors
+function RegularPrism(N::Integer; rInner = 0, rOuter = 1, zMin = -1, zMax = 1)
+    T = float(promote_type(typeof.((rInner, rOuter, zMin, zMax))...))
+    r = rInner == 0 ? T(rOuter) : T(rInner)..T(rOuter)
+    z = zMax == -zMin ? T(zMax) : T(zMin)..T(zMax)
+    RegularPrism(N, T, r, z)
+end
+RegularPrism(N::Integer, rInner, rOuter, zMin, zMax) = RegularPrism(N; rInner = rInner, rOuter = rOuter, zMin = zMin, zMax = zMax)
+
+function RegularPrism(N::Integer, rOuter::R, height::H) where {R<:Real, H<:Real}
+    T = float(promote_type(R,H))
+    RegularPrism( N, T, T(rOuter), T(height/2))
+end
+
+
+@inline in(p::CylindricalPoint, hp::RegularPrism{N, T, <:Real}) where {N,T} = begin
+    _in_z(p, hp.z) && abs(p.r * cos(T(π/N) - mod(p.φ, T(2π/N))) / cos(T(π/N))) <= hp.r
+end
+
+@inline in(p::CylindricalPoint, hp::RegularPrism{N, T, <:AbstractInterval}) where {N,T} = begin
+    _in_z(p, hp.z) && p.r * cos(T(π/N) - mod(p.φ, T(2π/N))) / cos(T(π/N)) in hp.r
+end
+
+@inline in(p::CartesianPoint, hp::RegularPrism) = in(CylindricalPoint(p), hp)
+
+# Special case: CartesianPoint in HexagonalPrism: use analytical formulas
+@inline in(p::CartesianPoint, hp::HexagonalPrism{T, <:Real}) where {T} =
+    _in_z(p, hp.z) && _in_x(p, hp.r * sqrt(T(3))/2) && _in_y(p, hp.r - abs(p.x)/sqrt(T(3)))
+    
+@inline in(p::CartesianPoint, hp::HexagonalPrism{T, <:AbstractInterval{T}}) where {T} = 
+    _in_z(p, hp.z) && abs(p.y) <= hp.r.right * sqrt(T(3))/2 && 
+    (
+        abs(p.y) >= hp.r.left * sqrt(T(3))/2 && _in_x(p, hp.r.right * T(0.5)) || 
+        abs(p.x) in (hp.r.left - abs(p.y) /sqrt(T(3)))..(hp.r.right - abs(p.y)/sqrt(T(3)))
+    )
+
+
+# read-in
+function Geometry(::Type{T}, ::Type{P}, dict::Union{Dict{String,Any}, Dict{Any,Any}}, input_units::NamedTuple
+            ) where {T, P <: Union{TriangularPrism, QuadraticPrism, PentagonalPrism, HexagonalPrism}}
+    length_unit = input_units.length
+    r = parse_r_of_primitive(T, dict, length_unit)
+    z = parse_height_of_primitive(T, dict, length_unit)
+    return P(T, r, z)
+end
+
+get_r_limits(rp::RegularPrism) = (_left_radial_interval(rp.r), _right_radial_interval(rp.r))
+get_z_limits(rp::RegularPrism) = (_left_linear_interval(rp.z), _right_linear_interval(rp.z))
+
+function sample(rp::RegularPrism{N,T}, Nsamps::NTuple{3,Int} = (2,N,2))::Vector{CylindricalPoint{T}} where {N,T}
+    rMin::T, rMax::T = get_r_limits(rp)
+    zMin::T, zMax::T = get_z_limits(rp)
+    samples = [
+        CylindricalPoint{T}(r,φ,z)
+        for z in (Nsamps[3] ≤ 1 ? zMin : range(zMin, zMax, length = Nsamps[3]))
+        for φ in 0:2π/N:2π-π/N #ignore the N given in Nsamps and sample with N
+        for r in (Nsamps[1] ≤ 1 ? _right_radial_interval(rp.r) : range(rMin, rMax, length = Nsamps[1]))
+    ]
+end
```

Comment:
This looks to me like sampling a cylinder, or am I wrong?
I know we check the sample points afterwards if they are inside but this just to substract differences of primitives.

Reply:
You are right. However, this is intended:
`sample` functions for `VolumePrimitives` are used to get the important points for grid creation.
In fact, sampling a `RegularPrism` looks like sampling a `Tube`, just that we have a **fixed** step size of `2π/N`, i.e. all the corners of the `RegularPrism` are sampled.


![HexagonalPrism](https://user-images.githubusercontent.com/30291312/116732877-a1afe500-a9eb-11eb-94e6-6e55777bf18b.png)
![Tube](https://user-images.githubusercontent.com/30291312/116732894-a6749900-a9eb-11eb-885c-9800b29007ae.png)


For contact painting, we will use the `sample` methods for the respective `SurfacePrimitives` which are not yet implemented.

Reply:
However this will break and sample a cylinder with `Nsamps = (2,x,2)` with `x > N`. I know sample is used to get the important points of the primitive, but it should also be able to sample the actual volume

Reply:
I suggest using `Nsamps::NTuple{3,Int} = (2,N+1,2)` and `for φ in (Nsamps[2] ≤ 1 ? 0 : range(0, 2π, length = Nsamps[2]))` with `CylindricalPoint{T}(r*cos(π/N)/cos(π/N - mod(φ, 2π/N)),φ,z)`

Reply:
I like this! Sampling with your code works now as intended:
![irregular_sampling](https://user-images.githubusercontent.com/30291312/116975134-6f3e0a80-acbf-11eb-8bc1-a8f72b863a70.png)